### PR TITLE
Fix React Native Compatibility

### PR DESCRIPTION
**Pull request by sirine-berguiga**
Hi and thank you for this library

I've tried to use it on my React Native project, but the hardcoded div is a problem. 

Here's a patch to enable customizing the error, while still defaulting to what you had (no BC break)

Enjoy!

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -115,10 +115,12 @@ export function configureAuth<
     children,
     renderLoading,
     renderUnauthenticated,
+    renderError = (status) => <div>Unhandled status: {status}</div>,
```
Comment (alan2207) [inline src/index.tsx:118]: Let's change it to the following to be able to use the error value:
`renderError = (error: Error) => <>{JSON.stringify(error)}</>,`

File: src/index.tsx [hunk 2]
```diff
@@ -115,10 +115,12 @@ export function configureAuth<
...
   }: {
     children: React.ReactNode;
     renderLoading: () => JSX.Element;
     renderUnauthenticated?: () => JSX.Element;
+    renderError?: (status: string) => JSX.Element;
```
Comment (alan2207) [inline src/index.tsx:123]: `renderError?: (error: Error) => JSX.Element;`

File: src/index.tsx [hunk 3]
```diff
@@ -133,7 +135,7 @@ export function configureAuth<
       return renderLoading();
     }
 
-    return <div>Unhandled status: {status}</div>;
+    return renderError(status);
```
Comment (alan2207) [inline src/index.tsx:138]: Here we can render the error:

To do that, we can replace the following:

```tsx
return renderError(status);
```
With the following:
```ts
if (status === 'error') {
  return renderError(error);
}

return null;
```

**Review by alan2207 — CHANGES_REQUESTED** [review]
Hey @sirine-berguiga , thanks a lot for the PR. I thought I had already removed the divs, but somehow missed this. Please check out the changes to make it even better.

**Comment by sirine-berguiga** [comment]
Hi @alan2207 ,
I did the requested changes and i added a note to the readme file.

Feel free to update the wording as you see fit.

**Review by alan2207 — APPROVED** [review]

**Comment by alan2207** [comment]
@sirine-berguiga Thanks!

**Pull request merged**